snyk/cli
# feat: pkg id provenance labels

## Pull Request Submission Checklist

## What does this PR do?
Add 'pkgIdProvenance' labels to dependency graph nodes when the package identity has been changed from what has been discovered in the manifest files.

This can happen in ecosystems like Python where package names are case insensitive, and Snyk needs to normalize them to match vulnerabilities.

This new label allows users to see what the package was originally called.

## How should this be manually tested?

```
snyk test --print-deps --json-file-output=out.json
cat out.json | jq.depGraph
```

On a Python project where package name gets change, for example if using Django instead of django.
On each node where the package name has changed there is now a label
```json
{
  "nodeId": "django@5.1.4",
  "pkgId": "django@5.1.4",
  "info": {
    "labels": {
      "pkgIdProvenance": "Django@5.1.4"
    }
  }

}
```

Or a Gradle project when using --gradle-normalize-deps.
```json
{
  "nodeId": "org.apache.logging.log4j:log4j-core@2.24.3",
  "pkgId": "org.apache.logging.log4j:log4j-core@2.24.3",
  "info": {
    "labels": {
      "pkgIdProvenance": "my.logging:log4j-core@2.24.3"
    }
  }

}
```

See plugin changes:
* https://github.com/snyk/snyk-gradle-plugin/pull/297
* https://github.com/snyk/snyk-python-plugin/pull/252
